src/ring_theory/polynomial/pochhammer.lean

```diff
@@ -132,11 +132,27 @@ end semiring
 section comm_semiring
 variables {S : Type*} [comm_semiring S]
 
-lemma pochhammer_succ_eval (n : ℕ) (k : S) :
+lemma pochhammer_succ_eval_aux (n : ℕ) (k : S) :
   (pochhammer S n.succ).eval k = (pochhammer S n).eval k * (k + ↑n) :=
 by rw [pochhammer_succ_right, polynomial.eval_mul, polynomial.eval_add, polynomial.eval_X,
     polynomial.eval_nat_cast]
 
+-- should `S` be explicit?
+lemma map_eval_pochhammer_eq_self {S : Type*} [semiring S] : ∀ (n : ℕ),
+  map (nat.cast_ring_hom S) (eval X (pochhammer ℕ[X] n)) = pochhammer S n
+| 0       := by simp
+| (n + 1) := by simp [pochhammer_succ_eval_aux, pochhammer_succ_right, map_eval_pochhammer_eq_self]
+
+lemma pochhammer_succ_eval {S : Type*} [semiring S] (n : ℕ) (k : S) :
+  (pochhammer S n.succ).eval k = (pochhammer S n).eval k * (k + ↑n) :=
+begin
+  rw [← map_eval_pochhammer_eq_self, ← map_eval_pochhammer_eq_self n, pochhammer_succ_eval_aux,
+    mul_add, polynomial.map_add, eval_add, mul_add],
+  congr,
+  { simp only [polynomial.map_mul, map_X, eval_mul_X] },
+  { simp [← nat.cast_comm] }
+end
```

Comment:
```suggestion
-- move me, add `left` version
lemma _root_.commute.cast_nat_right {S : Type*} [semiring S] (k : S) (n : ℕ) :
  commute k n :=
by simpa only [mul_one] using (commute.one_right k).cast_nat_mul_right n

lemma pochhammer_succ_eval {S : Type*} [semiring S] (n : ℕ) (k : S) :
  (pochhammer S n.succ).eval k = (pochhammer S n).eval k * (k + ↑n) :=
begin
  simp only [← pochhammer_map (nat.cast_ring_hom S), pochhammer_succ_right, polynomial.map_mul,
    polynomial.map_add, map_X, polynomial.map_nat_cast, mul_add, eval_add, eval_mul_X,
    C_eq_nat_cast, (commute.cast_nat_right _ n).eq, ← eval_C_mul],
  rw (commute.cast_nat_right _ n).eq,
end
```

Reply:
Actually, due to various defeqs, your proof can be golfed to
```
lemma _root_.commute.cast_nat_right {S : Type*} [semiring S] (k : S) (n : ℕ) :
  commute k n :=
(nat.cast_comm _ _).symm
```